modules/programs/zsh.nix

```
@@ -60,6 +60,15 @@ let
         description = "History file location";
       };
 
+      ignorePattern = mkOption {
+        type = types.nullOr types.string;
```

```suggestion
        type = types.nullOr types.envVar;
```

According to a stackoverflow answer, multiple patterns can be used with by separating them with a `:`.

---

No it doesn't. From the man page:
> Note that HISTORY_IGNORE defines a single pattern: to specify alternatives use the `(first|second|...)' syntax.

I did a test of `:` and it really didn't work.

---

Thanks, in that case we can replace the `types.string` with either `types.seperatedString "|"` or a list of string and we concat ourself in the config block.

---

Is there something like a `type.regex`? I think that would be ideal since it appears that this pattern is basically a regex.

---

The advantage of using a list or `types.separatedString` is that the module system will be support merging values whereas `types.regex` does not AFAIK.